modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/transformation/EventUtils.scala

```diff
@@ -138,8 +138,18 @@ object EventUtils {
     * @param instance self-describing JSON that needs to be transformed
     * @return list of columns or flattening error
     */
-  def flatten[F[_]: Monad: RegistryLookup: Clock](resolver: Resolver[F], instance: SelfDescribingData[Json]): EitherT[F, FailureDetails.LoaderIgluError, List[String]] =
-    getOrdered(resolver, instance.schema).map { ordered => FlatData.flatten(instance.data, ordered, getString, NullCharacter) }
+  def flatten[F[_]: Monad: RegistryLookup: Clock](resolver: Resolver[F], lookup: LookupProperties[F], instance: SelfDescribingData[Json]): EitherT[F, FailureDetails.LoaderIgluError, List[String]] = {
+    val fetchProperties = getOrdered(resolver, instance.schema).map { ordered =>
+      FlatSchema.extractProperties(ordered)
+    }.flatTap { props =>
+      EitherT.right(lookup.put(instance.schema, props))
```

Comment:
There is also some parsing/fetching in `Flattening`, does it make sense to cache sth there?